### ISSUE TITLE
Calendar picker questions display text in English

### ISSUE THREAD

**Issue by cusackjm** [episphere/connect#1027]
![image](https://github.com/episphere/connect/assets/85582895/5394875c-fcd7-43d4-b642-831a4d9acb19)
![image](https://github.com/episphere/connect/assets/85582895/271e27b8-5870-4b2b-bd61-c9ac81a41cdd)

Calendar pickers questions display text in English (Blood/Urine/Mouthwash, Blood/Urine, COVID-19, Menstrual Cycle Surveys)

**Comment by robertsamm** [comment]
@Davinkjohnson @sonyekere can someone on DevOps confirm whether this is something that can be fixed?

**Comment by cunnaneaq** [comment]
Thanks, @bransteitterbr! @m-j-horner and @depietrodeanna do you think it's OK for the calendar picker to be in English for the biospecimen surveys even when the participant selects Spanish?  I think our other option is to find another way to ask for the date and re-release the surveys.  Please let me know what you think!

**Comment by depietrodeanna** [comment]
@cunnaneaq @m-j-horner this is tough. This is not optimal but I'm not sure where fixing it falls in the line of priority items. For most months, the Spanish is just _close_ enough to English that folks can still make an educated guess but that's not the ideal scenario for accessibility.

**Comment by robertsamm** [comment]
@depietrodeanna on the PWA where we state something like "once you start a survey in Spanish you must remain in Spanish", could we add a statement that states that we recommend they change their device settings to Spanish? (and not just toggle to Spanish in the PWA)

**Comment by depietrodeanna** [comment]
That's possible. I could also ask for recommendation from OCPL on the UX. We would need to provide instructions somewhere and probably add a KBA for this as well @robertsamm

**Comment by robertsamm** [comment]
@depietrodeanna Michelle agrees with your suggestion to ask OCPL if they have any suggestions on the UI

**Comment by depietrodeanna** [comment]
I'll make a note to discuss with them and follow up here.

**Comment by depietrodeanna** [comment]
Hi @bransteitterbr and @sonyekere. We discussed with OCPL and the ideal UX is not to ask the user to change their computer settings to Spanish if the only issue persisting in English is the calendar picker (this creates undue user burden). Could your team reassess among the developers to see if anyone is able to continue troubleshooting this? The hope is that we can find a fix our Dev team can implement rather than a fix that requires action by the user. Let me know if you have further questions or want to discuss.

**Comment by sonyekere** [comment]
Hi @depietrodeanna , just wanted to share a quick update on the calendar picker item!

I checked in with Davin and Brian and at this point, supporting multiple languages within the existing calendar picker would require migrating the entire frontend to a framework. That would be a pretty large effort and may introduce some  dependencies that DevOps is trying to avoid.

A potential alternative that Brian suggested is to move to dropdown fields for Month, Day, and Year. This would give us more control over the displayed text, allowing us to localize the month names (e.g., in Spanish) without requiring users to change their device settings. It would be a simpler, less visual experience than the current calendar picker, but it could help address the language issue more directly. We can definitely consider this further if the team would like to consider this change in the next release.

**Comment by depietrodeanna** [comment]
That would be fine with me, but for now let's press pause until I touch base with BSWH. I'm going to ask a Spanish speaking team member to go through one of the surveys that has a calendar picker (@cunnaneaq could you let me know which survey you'd recommend they go through in MyConnect?) in Stage and see what the user experience is like/if they think this is worth the effort of Dev changing the structure and eliminating the picker.

**Comment by cunnaneaq** [comment]
thanks all!  @depietrodeanna - calendar picker questions are used in Blood/Urine/Mouthwash, Blood/Urine, COVID-19, and Menstrual Cycle Surveys

**Comment by depietrodeanna** [comment]
@robertsamm @brotzmanmj @Davinkjohnson BSWH provided the following in UX testing:

_Although it is in English, she stated that “it is generally understood that participants will look at the calendar and select the correct dates, even if it’s in English.” She did mention, however, that if feasible, it would be helpful to update the picker to better match user preferences—particularly options such as “Clear” and “This Month.” That said, she does not believe this is a major issue._ 

It seems this is not high priority and can be pushed back in the DevOps queue.

**Comment by Davinkjohnson** [comment]
@depietrodeanna @cunnaneaq  If we keep this open, the only way we really have to address it is to go with the suggestion to change away from the calendar picker. "_move to dropdown fields for Month, Day, and Year._" 

Do we have a decision from your team that this is what you'd like?

**Comment by depietrodeanna** [comment]
I don't have a strong preference either way. If we think calendars are easier to navigate (even if in English when preferred language is Spanish), then I'm OK keeping the calendars rather than shifting to dropdown. Given BSWH's feedback it seems that this is not a high priority shift. @brotzmanmj and @cunnaneaq let me know if you agree.

**Comment by cunnaneaq** [comment]
I agree with Deanna.  I remember when the calendars were implemented, we thought they would be more user-friendly.  Maybe we leave them as is in the current biospecimen surveys, but going forward we phase out the use of calendar pickers since we don't have as much control over the language preference.

**Comment by sonyekere** [comment]
Per Deanna (6/17), the final decision was not to retroactively replace any calendar pickers with dropdowns. Going forward, calendar pickers will be phased out in favor of dropdowns for date selection.

**Comment by sonyekere** [comment]
@depietrodeanna based on the feedback not to retroactively replace the existing calendar pickers, would it make sense to close this issue? If there's a need to implement dropdown date fields in the future, we can create a new issue to track that work.

**Comment by depietrodeanna** [comment]
yep @sonyekere
